platforms/android/tangram/src/main/cpp/androidPlatform.cpp

```diff
@@ -385,65 +359,52 @@ void setCurrentThreadPriority(int priority) {
     setpriority(PRIO_PROCESS, 0, priority);
 }
 
-void labelPickCallback(jobject listener, const Tangram::LabelPickResult* labelPickResult) {
+void AndroidPlatform::labelPickCallback(const LabelPickResult* labelPickResult) {
 
     JniThreadBinding jniEnv(jvm);
 
-    float position[2] = {0.0, 0.0};
-
-    jobject labelPickResultObject = nullptr;
+    float x = 0.f, y = 0.f;
+    double lng = 0., lat = 0.;
+    int type = 0;
+    jobject hashmap = nullptr;
 
     if (labelPickResult) {
         auto properties = labelPickResult->touchItem.properties;
 
-        position[0] = labelPickResult->touchItem.position[0];
-        position[1] = labelPickResult->touchItem.position[1];
+        x = labelPickResult->touchItem.position[0];
+        y = labelPickResult->touchItem.position[1];
 
-        jobject hashmap = jniEnv->NewObject(hashmapClass, hashmapInitMID);
+        hashmap = jniEnv->NewObject(hashmapClass, hashmapInitMID);
```

Comment:
Just FYI, I was curious on if the allocations here are reflected aptly in the profiler, seems to be the case and gets deallocated aptly during onLabelPick implementation.